src/dstore/pmix_esh.c

```diff
@@ -228,22 +491,64 @@ int _esh_init(pmix_info_t info[], size_t ninfo)
         return PMIX_SUCCESS;
     }
     else {
+        int rc = 0;
+        int tbl_idx;
+        nspace_tbl_entry_t *nspace_tbl_entry = NULL;
+
+        if (NULL != info) {
```

Comment:
Why do you separate this for the server and client? I thinkg that this loop has to be at the beginning of the function and it has to be uniform for both client and server (with necessary distinct's)